bugwarrior/services/bts.py

```diff
@@ -0,0 +1,175 @@
+import debianbts
+import requests
+
+from bugwarrior.config import die, asbool
+from bugwarrior.services import Issue, IssueService, ServiceClient
+
+import logging
+log = logging.getLogger(__name__)
+
+UDD_BUGS_SEARCH = "https://udd.debian.org/bugs/"
+
+class BTSIssue(Issue):
+    SUBJECT = 'btssubject'
+    URL = 'btsurl'
+    NUMBER = 'btsnumber'
+    PACKAGE = 'btspackage'
+    SOURCE = 'btssource'
+    FORWARDED = 'btsforwarded'
+
+    UDAS = {
+        SUBJECT: {
+            'type': 'string',
+            'label': 'Debian BTS Subject',
+        },
+        URL: {
+            'type': 'string',
+            'label': 'Debian BTS URL',
+        },
+        NUMBER: {
+            'type': 'numeric',
+            'label': 'Debian BTS Number',
+        },
+        PACKAGE: {
+            'type': 'string',
+            'label': 'Debian BTS Package',
+        },
+        SOURCE: {
+            'type': 'string',
+            'label': 'Debian BTS Source Package',
+        },
+        FORWARDED: {
+            'type': 'string',
+            'label': 'Debian BTS Forwarded URL',
+        },
+    }
+    UNIQUE_KEY = (URL, )
+
+    PRIORITY_MAP = {
+        'wishlist': 'L',
+        'minor': 'L',
+        'normal': 'M',
+        'important': 'M',
+        'serious': 'H',
+        'grave': 'H',
+        'critical': 'H',
+    }
+
+    def to_taskwarrior(self):
+        return {
+            'priority': self.get_priority(),
+
+            self.URL: self.record['url'],
+            self.SUBJECT: self.record['subject'],
+            self.NUMBER: self.record['number'],
+            self.PACKAGE: self.record['package'],
+            self.SOURCE: self.record['source'],
+            self.FORWARDED: self.record['forwarded'],
+        }
+
+    def get_default_description(self):
+
+        return self.build_default_description(
+            title=self.record['subject'],
+            url=self.get_processed_url(self.record['url']),
+            number=self.record['number'],
+            cls='issue'
+        )
+
+    def get_priority(self):
+        return self.PRIORITY_MAP.get(
+            self.record.get('severity'),
+            self.origin['default_priority']
+        )
+
+
+class BTSService(IssueService, ServiceClient):
+    ISSUE_CLASS = BTSIssue
+    CONFIG_PREFIX = 'bts'
+
+    def __init__(self, *args, **kw):
+        super(BTSService, self).__init__(*args, **kw)
+        self.bts = debianbts
```

Comment:
My bad, I think the target is relative to the module in which the patching is occurring. So it would be `with mock.patch('bts.debianbts', FakeBtsLib):`